src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -12,36 +12,50 @@
 from builtins import open, zip
 from contextlib import contextmanager
 
-from future.utils import PY3, raise_with_traceback
-from setproctitle import setproctitle as set_process_title
-
 from pants.base.build_environment import get_buildroot
 from pants.base.exception_sink import ExceptionSink, SignalHandler
 from pants.base.exiter import PANTS_FAILED_EXIT_CODE, PANTS_SUCCEEDED_EXIT_CODE, Exiter
 from pants.bin.local_pants_runner import LocalPantsRunner
+from pants.init.logging import encapsulated_global_logger
 from pants.init.util import clean_global_runtime_state
 from pants.java.nailgun_io import NailgunStreamStdinReader, NailgunStreamWriter
-from pants.java.nailgun_protocol import ChunkType, NailgunProtocol
-from pants.pantsd.process_manager import ProcessManager
+from pants.java.nailgun_protocol import ChunkType, MaybeShutdownSocket, NailgunProtocol
 from pants.util.contextutil import hermetic_environment_as, stdio_as
 from pants.util.socket import teardown_socket
 
 
 class DaemonSignalHandler(SignalHandler):
```

Comment:
I don't think this signal handler ever takes effect, because we don't have a good way to enable it from the main thread (this whole piece of code runs in a child thread, and we can't use custom signal handlers in those for now). I think this entire class can be removed.